### PR TITLE
minor: Implement common traits for OneOrManyWithParens

### PR DESCRIPTION
**Pull request by gstvg**
This allows accessing all inner values without matching

### CODE REVIEW

**Review by jmhain — COMMENTED** [review]
lgtm, cc @alamb

**Review by jmhain — APPROVED** [review]

**Review by alamb — COMMENTED** [review]
Thanks @gstvg  -- can you please add a test showing what this contribution allows? That way we won't accidentally break it in the future

**Comment by alamb** [comment]
Marking as draft as I think this PR is no longer waiting on feedback. Please mark it as ready for review when it is ready for another look

**Comment by gstvg** [comment]
> Thanks @gstvg -- can you please add a test showing what this contribution allows? That way we won't accidentally break it in the future

Sure, it makes sense, so much so that while I was writing it, I realized it wasn't enough for my intended usage 😅
I ended up implementing 3 more traits
Thanks @jmhain and @alamb

File: src/ast/mod.rs
Comment (gstvg) [inline src/ast/mod.rs:1049]: This is a bit strange, it could be 
1) `vec![one]` and just return `Vec::into_iter`, but it allocates, or
2) create and implement a new iterator, but it's much more work, and we won't get for free additional iterator traits impls.

Just let me know if other approach is preferable

Comment (alamb) [reply src/ast/mod.rs:1049]: > create and implement a new iterator, but it's much more work, and we won't get for free additional iterator traits impls.

this is what i would have expected. 

Not sure what you mean "won't get for free additional iterator traits impl"

Another potential approach is returning a `Box`ed iterator

File: src/ast/mod.rs
Comment (alamb) [inline src/ast/mod.rs:1008]: This is looking good -- I think if we are going to add this code we should also add examples to the documentation f `OneOrManyWithParens` otherwise it will likely be hard to find

So like 
```rust 
/// # Example: accessing as a slice:
/// ...
```

```rust 
/// # Example: iterating
/// ...
```

**Review by alamb — COMMENTED** [review]
Thanks @gstvg  -- this is looking close

File: src/ast/mod.rs
Comment (gstvg) [reply src/ast/mod.rs:1027]: Done!

File: src/ast/mod.rs
Comment (gstvg) [reply src/ast/mod.rs]: It wasn't too much work after all :grimacing: 

> Not sure what you mean "won't get for free additional iterator traits impl"

It's actually just `DoubleEndedIterator`, I thought there were more

**Review by alamb — APPROVED** [review]
This looks good to me -- thank you @gstvg  and I apologize for the delay in reviewing

**Comment by alamb** [comment]
Thanks again @gstvg

**Comment by gstvg** [comment]
Thanks @jmhain and @alamb :pray: 

> This looks good to me -- thank you @gstvg and I apologize for the delay in reviewing

It's okay, don't worry. If I ever need something to be reviewed with priority, I will ping you or another maintainer

**Pull request merged**
